addons/hw_drivers/driver.py

```diff
@@ -47,29 +52,41 @@ def supported(cls, device):
         """
         return False
 
-    def action(self, data):
+    def action(self, action='', **kwargs):
         """Helper function that calls a specific action method on the device.
 
-        :param data: the `_actions` key mapped to the action method we want to call
-        :type data: string
+        :param str action: the action to be executed
+        :param dict kwargs: the parameters to be passed to the action method
+        :return: the result of the action method
         """
-        self._actions[data.get('action', '')](data)
+        try:
+            response = {'status': 'success', 'result': self._actions[action](**kwargs), 'action_args': {**kwargs}}
+        except Exception as e:
+            _logger.exception("Error while executing action")
```

Comment:
more details can be given in the exception string like the action and the arguments (it's always better than reading the exception traceback)